### PR TITLE
Added site recovery resource for DRaaS

### PR DESCRIPTION
**Pull request by saloneerege**

### CODE REVIEW

File: examples/variables.tf [hunk 1]
```diff
@@ -51,10 +51,25 @@ variable storage_capacity {
 
 variable num_hosts {
   description = "The number of hosts."
-  default     = ""
+  default     = 1
 }
 
 variable provider_type {
   description = "Determines what additional properties are available based on cloud provider. Acceptable values are ZEROCLOUD and AWS with AWS as the default value."
```
Comment (rchadalawada) [inline examples/variables.tf]: I thought you are removing the references to ZEROCLOUD

Comment (saloneerege) [reply examples/variables.tf]: Already removed in master branch.

File: vmc/resource_site_recovery.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+				ForceNew:    true,
+				Required:    true,
+				Description: "SDDC identifier",
+			},
+			"srm_extension_key_suffix": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 13),
+				Description:  "Custom extension key suffix for SRM. If not specified, default extension key will be used. The custom extension suffix must contain 13 characters or less, be composed of letters, numbers, ., - characters only. The extension suffix must begin and end with a letter or number. The suffix is appended to com.vmware.vcDr- to form the full extension key",
```
Comment (shylasrinivas) [inline vmc/resource_site_recovery.go:43]: is such a long description needed here? will help to have this in the docs..

File: vmc/resource_site_recovery_test.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
+/* Copyright 2019 VMware, Inc.
```
Comment (shylasrinivas) [inline vmc/resource_site_recovery_test.go:1]: update copyright

File: vmc/resource_site_recovery.go [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+}
+
+func resourceSiteRecoveryRead(d *schema.ResourceData, m interface{}) error {
+	connector := (m.(*ConnectorWrapper)).Connector
+	sddcID := d.Id()
+	orgID := (m.(*ConnectorWrapper)).OrgID
+	siteRecoveryClient := draas.NewDefaultSiteRecoveryClient(connector)
+	siteRecovery, err := siteRecoveryClient.Get(orgID, sddcID)
+	if err != nil {
```
Comment (shylasrinivas) [inline vmc/resource_site_recovery.go:126]: do we plan re-create resource if it is not found? Eg: case when its deleted via console accidentally

File: vmc/resource_vmc_public_ip.go [hunk 5]
```diff
@@ -86,6 +89,11 @@ func resourcePublicIpRead(d *schema.ResourceData, m interface{}) error {
 	if len(uuid) > 0 {
 		publicIp, err := nsxVmcAwsClient.GetPublicIp(uuid)
 		if err != nil {
+			if err.Error() == errors.NewNotFound().Error() {
```
Comment (shylasrinivas) [inline vmc/resource_vmc_public_ip.go:92]: why is this showing up as a change? it is already present in master.

File: website/docs/r/site_recovery.html.markdown [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+layout: "vmc"
+
+page_title: "VMC: vmc_site_recovery"
+sidebar_current: "docs-vmc-resource-site-recovery"
+
+description: |-
+  Provides a resource to manage site recovery.
+---
+
+# vmc_public_ip
```
Comment (shylasrinivas) [inline website/docs/r/site_recovery.html.markdown:11]: supposed to be vmc_site_recovery?

File: website/docs/r/site_recovery.html.markdown [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+The following arguments are supported for vmc_public_ip resource:
+
+* `sddc_id` - (Required) SDDC Identifier.
+
+* `srm_extension_key_suffix` - (Optional) Custom extension key suffix for SRM. If not specified, default extension key will be used. The custom extension suffix must contain 13 characters or less, be composed of letters, numbers, ., - characters only. The extension suffix must begin and end with a letter or number. The suffix is appended to com.vmware.vcDr- to form the full extension key.        
+   Length of srm_extension_key_suffix cannot be more than 13 characters.                                   
+
+## Attributes Reference
+
+In addition to arguments listed above, the following attributes are exported after public IP creation:
```
Comment (shylasrinivas) [inline website/docs/r/site_recovery.html.markdown:43]: Below section refers to public IP instead of site recovery.

**Pull request closed without merging**